addon/adapters/cloud-firestore-modular.ts

```diff
@@ -164,7 +165,7 @@ export default class CloudFirestoreModularAdapter extends Adapter {
         if (docSnapshot.exists()) {
           resolve(flattenDocSnapshot(docSnapshot));
         } else {
-          reject(new Error(`Record ${id} for model type ${type.modelName} doesn't exist`));
+          reject(new FirebaseError('not-found', `Record ${id} for model type ${type.modelName} doesn't exist`));
```

Comment:
The reason why I was hesitant to do this was because it's confusing if we use `FirebaseError` because it's **not** a firebase error. If you do a `getDoc()` and the document doesn't exist, it doesn't really throw an error that it doesn't exist. It still responds successfully but the `snapshot.exists` would be false.

So this is really an error from adapter-level for alignment on how ember-data expects to work.

I've never played around with error messages before, but I assume that we can probably created our own `FirebaseError` class and name it maybe something like `AdapterError`. But I truly don't know what the advantage of that over regular error message.

Reply:
I kept the issue open because I thought that there might be a better way to handle errors, but it's just low of my bucket list right now.

Reply:
I'm really just looking for a cleaner solution to handle the rejected scenario specifically for missing records without having to regex the error message for the current "... doesn't exist" message.

Reply:
I'll only be able to check this out again by May. In the meantime, I welcome PRs that implements a class that extends `Error` so we could do something like this:

```javascript
try {
  await this.store.findRecord('user', 'user_1');
} catch (error) {
  if (error instanceof AdapterRecordNotFoundError) {
    // ...
  }
}
```

Also, I think [`String.endsWith`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith) would be useful rather than trying to use regex.